planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp

```diff
@@ -712,6 +728,28 @@ void GoalPlannerModule::setStopPath(BehaviorModuleOutput & output)
   }
 }
 
+void GoalPlannerModule::setStopPathInCurrentPath(BehaviorModuleOutput & output)
+{
+  if (status_.prev_is_safe_dynamic_objects || status_.prev_stop_path_after_approval == nullptr) {
+    // safe -> not_safe or no prev_stop_path: generate new stop_path
+    output.path = std::make_shared<PathWithLaneId>(generateStopPointInCurrentPath());
+    output.reference_path = getPreviousModuleOutput().reference_path;
+    status_.prev_stop_path_after_approval = output.path;
+    // set stop path as pull over path
+    mutex_.lock();
```

Comment:
will replace with `std::lock_guard<std::mutex> lock(mutex_);`

Reply:
changed in https://github.com/autowarefoundation/autoware.universe/pull/4927/commits/dbc6b460638d80ae3c6c596ce2e45b6fca964be1